### PR TITLE
fix memory leak and remove deprecated factor 0.5

### PR DESCRIPTION
**Pull request by hvanhaev**

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
Since you are cleaning the code here, please:
- If you plan to use it in production, and not only in tests as now, make this thread safe: stream module
- Either remove line https://github.com/hvanhaev/cmssw/blob/2fa8b0f261cbc16171615447ee7e835833bd49a3/RecoLocalCalo/Castor/src/RecHitCorrector.cc#L117 , or replace the thread-unsafe cout with some LogError/LogWarning
- remove all commented out lines with cout's
- Since you are there, you could also replace the for loops in the produce() method with range based loops

**Comment by hvanhaev** [comment]
Dear Andrea,

Thank you for the feedback!
I will try to implement your proposals to clean it up.

Cheers,
Hans


On 16 Jan 2018, at 14:40, perrotta <notifications@github.com<mailto:notifications@github.com>> wrote:


Since you are cleaning the code here, please:

  *   If you plan to use it in production, and not only in tests as now, make this thread safe: stream module
  *   Either remove line https://github.com/hvanhaev/cmssw/blob/2fa8b0f261cbc16171615447ee7e835833bd49a3/RecoLocalCalo/Castor/src/RecHitCorrector.cc#L117 , or replace the thread-unsafe cout with some LogError/LogWarning
  *   remove all commented out lines with cout's
  *   Since you are there, you could also replace the for loops in the produce() method with range based loops

File: RecoLocalCalo/Castor/src/RecHitCorrector.cc [hunk 1]
```diff
@@ -163,11 +163,14 @@ RecHitCorrector::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 	if (ok) {
 	    CastorRecHit *correctedhit = new CastorRecHit(rechit.id(),correctedenergy,time);
```
Comment (VinInn) [inline RecoLocalCalo/Castor/src/RecHitCorrector.cc]: why all this?
rec.emplace_back(rechit.id(),correctedenergy,time)
should suffice

**Comment by hvanhaev** [comment]
Is this emplace_back something newly introduced or did I miss it many years ago?
Sorry I was not aware of it…

Cheers,
Hans



On 16 Jan 2018, at 18:33, Vincenzo Innocente <notifications@github.com<mailto:notifications@github.com>> wrote:


@VinInn commented on this pull request.

________________________________

In RecoLocalCalo/Castor/src/RecHitCorrector.cc<https://github.com/cms-sw/cmssw/pull/21860#discussion_r161830599>:

> @@ -163,11 +163,14 @@ RecHitCorrector::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
        if (ok) {
            CastorRecHit *correctedhit = new CastorRecHit(rechit.id<http://rechit.id>(),correctedenergy,time);


why all this?
rec.emplace_back(rechit.id<http://rechit.id>(),correctedenergy,time)
should suffice

**Comment by VinInn** [comment]
> On 16 Jan, 2018, at 9:58 PM, Hans Van Haevermaet <notifications@github.com> wrote:
> 
> Is this emplace_back something newly introduced or did I miss it many years ago?
in C++11, so ~6 years ago.
>  
> Sorry I was not aware of it… 

v.

**Comment by hvanhaev** [comment]
Ok, I wrote this 7 years ago ;-)
Thanks for the advice, I will implement it.

Cheers,
Hans

**Comment by perrotta** [comment]
By the way: this PR is submitted to CMSSW_8_0_X!
Do you really plan to update 80X with it? Isn't it foreseen for a private branch, maybe?
In any case, if you want it in the release, it has to be submitted and enter the master first

**Comment by perrotta** [comment]
-1
- I think that you can close this one, as you wrote in https://github.com/cms-sw/cmssw/pull/21860#issuecomment-358927726 . In any case, you should make a PR to the master first, with all the adjustment to the code that are not considered here yet: then one can discuss about possible backports

**Pull request closed without merging**